lib/sprinkler/devices/watering_policy.ex

```diff
@@ -0,0 +1,88 @@
+defmodule Sprinkler.Devices.WateringPolicy do
+  # TODO: check this values
+  @soil_moisture_wet 70
+  @soil_moisture_medium_wet 50
+  @ambience_temperature_high 25
+  @humidity_high 50
+
+  def should_irrigate(moisture_input, temperature_input, humidity_input) do
+    unless soil_moisture_wet?(moisture_input) do
+      when_soil_moisture_not_wet(moisture_input, temperature_input, humidity_input)
+    end
+  end
```

Comment:
You can return `:low`, `:medium` and `:high` instead of actually irrigating

Reply:
Alternatively, we could just return whether or not we should irrigate and not worry about medium or high. I mean we are going to be checking constantly so it doesn't make much sense to have something this complicated

Reply:
I understand your point, but I really think could be useful to have the different irrigation levels, we can discuss with the team today, wdyt?

Reply:
I agree with @brunvez that a first approach could be just returning whether to irrigate or not (since we'll be checking constantly I guess the level can be just a standart one every time). Then we can make the algorith more complex if necessary.

Reply:
following this, and seeing how complex is the resultant method, we can follow this approach and just water or not, are all agree?